src/config/configAggregator.ts

```diff
@@ -137,7 +138,7 @@ export class ConfigAggregator extends AsyncOptionalCreatable<ConfigAggregator.Op
    *
    * @param key The config key.
    */
-  public static getValue(key: string): ConfigInfo {
+  public static getValue(key: string): ConfigInfo | undefined {
```

Comment:
I think so.  Not in actual functionality but to a compiler, if they don't have skipLibChecks, then it might.

I could back this out and let the consumers stumbled into the potential `undefineds` the hard way, and save the type fix for the next major.

Reply:
I would consider that a breaking change, anything where you'd have to fix code to consume a new version 😦 

I wish we could introduce this now, but think we can only introduce type narrowing changes without a major 🙁 

we can leave a comment in there for now 🤷‍♂️ - could even link to this thread